providers/shodan/provider/provider.go

```diff
@@ -51,10 +52,9 @@ func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error)
 	if token == "" {
 		token = os.Getenv("SHODAN_TOKEN")
 	}
-	if token == "" {
-		return nil, errors.New("no shodan token provided, use --token or SHODAN_TOKEN")
```

Comment:
This is also validated in the connection. We don't need to check it here. This makes sure inventory files with vault secrets are accepted. If no token is provided, we will error out when the connection is created https://github.com/mondoohq/cnquery/blob/ffa713536aed31ebed30c94ed015ff9d5c7d95ee/providers/shodan/connection/connection.go#L53-L55